src/d3d8/d3d8_device.cpp

```diff
@@ -994,30 +994,46 @@ namespace dxvk {
     HRESULT res = GetD3D9()->CreateStateBlock(d3d9::D3DSTATEBLOCKTYPE(Type), &pStateBlock9);
 
     D3D8StateBlock* pStateBlock = new D3D8StateBlock(this, Type, pStateBlock9.ref());
-    m_stateBlocks.insert(pStateBlock);
 
-    *pToken = DWORD(reinterpret_cast<uintptr_t>(pStateBlock));
+    m_stateBlocks.insert({m_token, pStateBlock});
+    *pToken = m_token;
+    m_token++;
 
     return res;
   }
 
   HRESULT STDMETHODCALLTYPE D3D8Device::CaptureStateBlock(DWORD Token) {
-    return reinterpret_cast<D3D8StateBlock*>(Token)->Capture();
+    try {
+      return m_stateBlocks.at(Token)->Capture();
+    } catch (const std::out_of_range& e) {
```

Comment:
Please no try/catch. Please use `.find` and check if != end. Maybe add a helper to do that and return nullptr.

Reply:
Done. I'd like to stress that these are highly unlikely to ever be hit, I just thought it would be a good idea for us to be a bit more robust and not like... crash.